index.js

```diff
@@ -125,19 +122,14 @@ function escapeRegExp(s) {
     return s.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
 }
 
-function matchFromRegex(s, options) {
-  if (options.regExpOptions) {
-    return new RegExp(s.source, options.regExpOptions)
-  } else {
-    var flags = s.toString().replace(/\/[^\/].*\//, '')
-    // If there is no global flag then there can only be one match
-    if (flags.indexOf('g') < 0) {
-      options.limit = 1;
-    }
-    return new RegExp(s.source, flags)
+function matchFromRegex(regex, options) {
+  // If there is no global flag then there can only be one match
+  if (!regex.global) {
```

Comment:
The `global` is a property of a RegExp indicating if the global flag is set (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global).

Hence, I believe it's a simpler way of achieving the same thing?

It looks to be supported in node v0.10 and v0.12...